src/integrationTest/java/uk/gov/hmcts/reform/sendletter/e2e/EndToEndWithEncryptionEnabledTest.java

```diff
@@ -0,0 +1,141 @@
+
+public class EndToEndWithEncryptionEnabledTest {
+
+    @Autowired
+    private MockMvc mvc;
+
+    @Autowired
+    private LetterRepository repository;
+
+    @SpyBean
+    private AppInsights insights;
+
+    @After
+    public void cleanUp() {
+        // This test commits transactions to the database
+        // so we must clean up afterwards.
+        repository.deleteAll();
+    }
+
+    @Test
+    public void should_handle_old_letter_model() throws Throwable {
+        should_upload_letter_and_mark_posted(
+            post("/letters")
+                .header("ServiceAuthorization", "auth-header-value")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(readResource("letter.json"))
+        );
+    }
+
+    @Test
+    public void should_handle_new_letter_model() throws Throwable {
+        should_upload_letter_and_mark_posted(
+            post("/letters")
+                .header("ServiceAuthorization", "auth-header-value")
+                .contentType(MediaTypes.LETTER_V2)
+                .content(readResource("letter-with-pdf.json"))
+        );
+    }
+
+    private void should_upload_letter_and_mark_posted(MockHttpServletRequestBuilder request) throws Throwable {
+        try (LocalSftpServer server = LocalSftpServer.create()) {
+            mvc.perform(request)
+                .andExpect(status().isOk())
+                .andReturn();
+
+            // Wait for letter to be uploaded.
+            await().atMost(15, SECONDS).untilAsserted(
+                () -> assertThat(server.lettersFolder.listFiles()).as("No letters uploaded!").isNotEmpty()
+            );
+
+            // Generate Xerox report.
+            createXeroxReport(server);
+
+            // The report should be processed and the letter marked posted.
+            await().atMost(15, SECONDS).untilAsserted(
+                () -> assertThat(letterHasBeenPosted()).as("Letter not posted").isTrue()
+            );
+        }
+    }
+
+    private boolean letterHasBeenPosted() {
+        List<Letter> letters = repository.findAll();
+        return letters.size() == 1 && letters.get(0).getStatus() == LetterStatus.Posted;
+    }
+
+    private void createXeroxReport(LocalSftpServer server) throws IOException {
+        Stream<UUID> letterIds = Arrays.stream(server.lettersFolder.listFiles()).map(this::validateGettingLetterId);
+        XeroxReportWriter.writeReport(letterIds, server.reportFolder);
+    }
+
+    // Decrypt encrypted zip file so that we can confirm if we are able to decrypt the encrypted zip using private key
+    // returning the Letter's ID from the filename.
+    private UUID validateGettingLetterId(File file) {
+
+        try {
+            PgpDecryptionHelper.decryptFile(
+                Files.readAllBytes(file.toPath()),
+                getClass().getResourceAsStream("/privatekey.asc"),
+                "Password1".toCharArray()
```

Comment:
does the private key need to be passphrase protected?

Reply:
yes it needs to passphrase protected